apiserver/facades/controller/caasunitprovisioner/provisioner.go

```diff
@@ -535,17 +535,21 @@ func (a *Facade) updateStatus(params params.ApplicationUnitParams) (
 // source (typically a cloud update event) and merges that with the existing unit
 // data model in state. The passed in units are the complete set for the cloud, so
 // any existing units in state with provider ids which aren't in the set will be removed.
-func (a *Facade) updateUnitsFromCloud(app Application, scale *int, unitUpdates []params.ApplicationUnitParams) error {
+func (a *Facade) updateUnitsFromCloud(app Application, scale *int, generation *int64, unitUpdates []params.ApplicationUnitParams) error {
 	logger.Debugf("unit updates: %#v", unitUpdates)
 	if scale != nil {
 		logger.Debugf("application scale: %v", *scale)
+		if *scale > 0 && len(unitUpdates) == 0 {
+			// no ops for empty units because we can not determine if it's stateful or not in this case.
+			logger.Warningf("ignoring empty event for %q", app.Tag().String())
```

Comment:
ignoring empty k8s event

debug not warning as it's nothing something the user needs to care about or can do anything to fix

Reply:
changed to debugf